massbalancemachine/data_processing/get_topo_data.py

```diff
@@ -65,37 +65,61 @@ def get_topographical_features(
 
     # Based on the stake location, find the nearest point on the glacier with
     # recorded topographical features
-    _retrieve_topo_features(
-        data, glacier_directories, gdirs_gridded, grouped_stakes, voi
-    )
+    _retrieve_topo_features(data, glacier_directories, gdirs_gridded,
+                            grouped_stakes, voi)
 
     # Check if the dataframe is not empty (i.e. no points were found)
     if data.empty:
         raise ValueError(
             "DataFrame is empty, no stakes were found for the region of interest. Please check if your \n"
-            "RGIIDs are correct, and your coordinates are in the correct CRS."
-        )
+            "RGIIDs are correct, and your coordinates are in the correct CRS.")
 
     data.to_csv(output_fname, index=False)
 
     return data
 
 
+def get_glacier_mask(df:pd.DataFrame, custom_working_dir:str):
+    """Gets glacier xarray from OGGM and masks it over the glacier outline."""
+    
+    # Initialize the OGGM Config
+    _initialize_oggm_config(custom_working_dir)
+    
+    # Initialize the OGGM Glacier Directory, given the available RGI IDs
+    rgi_id = df.RGIId.unique()
+    gdirs = _initialize_glacier_directories(rgi_id)
+    
+    # Get oggm data for that RGI ID
+    for gdir in gdirs:
+        if gdir.rgi_id == rgi_id[0]:
```

Comment:
Are you sure that there will always be a single RGI and not multiple available?

Reply:
For now yes because this function creates the glacier dataframe for one glacier only (and so there is only one RGI in the `df`). In case of multiple glaciers to outline, the function should be called separately.